### PR TITLE
Validate incoming message

### PR DESCRIPTION
**Pull request by yenda**
First draft, will go back to it on Monday

- add validate method to StatusMessage protocol
- spec all message types for use in validate method
- use valid method after transit/decode step to reject
invalid messages

status: ready

### CODE REVIEW

File: src/status_im/transport/db.cljs [hunk 1]
```diff
@@ -35,3 +37,41 @@
...
+
+(spec/def :group-chat/admin :global/public-key)
+(spec/def :group-chat/participants (spec/coll-of :global/public-key :kind set?))
+(spec/def :group-chat/signature string?)
+
+(spec/def ::content (spec/conformer (fn [content]
+                                      (cond
+                                        (string? content) {:text content}
+                                        (map? content) content
+                                        :else :clojure.spec/invalid))))
```
Comment (janherich) [inline src/status_im/transport/db.cljs:58]: We should probably validate map content more throughly, dependent on whether it's `command` or `text/plain`/`emoji` content type, certain keys has to be there in order to not crash the rendering. 
For `text/plain` it's just `:text` which has to be string (can by simply conformed to string by calling `(update content :text str)`, for command, it's little bit more complicated and I will provide you with map of required keys/types-of-keys.

**Review by janherich — CHANGES_REQUESTED** [review]

**Review by mandrigin — CHANGES_REQUESTED** [review]
Can we also add unit tests to this functionality? Looks like it can have a good ratio of effort vs result.

File: src/status_im/transport/db.cljs (hunk 1 above)
Comment (janherich) [reply src/status_im/transport/db.cljs:58]: So the spec:

`content-type`:`text/plain`
It has to be a map with required key `:text` of type string, optional key `:response-to` of type string as well.

`content-type`:`command`
It has to be a map with required keys `:command-path` of type `[string set(keywords or strings)]`  and `:params` of type `map(keyword->anything)`

**Comment by lukaszfryc** [comment]
@yenda is there a way for me to send custom messages using a bot or something? I'd like to check different payloads to avoid recent [issue that was crashing the app](https://status-im.slack.com/archives/C88MZP11P/p1538472928000100).

cc @mandrigin

**Comment by janherich** [comment]
@lukaszfryc I have some Clojure repl scripts to do that if you have dev environment running, or @cammellos maybe even has something shell based.

File: src/status_im/transport/db.cljs (hunk 1 above)
Comment (yenda) [reply src/status_im/transport/db.cljs:58]: done

**Comment by yenda** [comment]
@mandrigin waiting for @gnl generative tests for that

**Review by janherich — APPROVED** [review]

**Comment by mandrigin** [comment]
@yenda can we just have a list of JSONs to test for now?

**Comment by gnl** [comment]
@mandrigin Proper gen-testing imminent, gimme a few minutes.

**Review by gnl — APPROVED** [review]

**Review by mandrigin — APPROVED** [review]

**Pull request merged**